corehq/util/soft_assert/api.py

```diff
@@ -3,9 +3,18 @@
 from corehq.apps.hqwebapp.tasks import send_mail_async, mail_admins_async
 from corehq.util.global_request import get_request
 from corehq.util.soft_assert.core import SoftAssert
+from django.views.debug import get_exception_reporter_filter
 
 
 def _send_message(info, backend):
+    try:
+        request = get_request()
+        # de-sensitize requests that might have passwords
+        filter = get_exception_reporter_filter(request)
+        request_repr = filter.get_request_repr(request)
```

Comment:
I couldn't test this (hard to reproduce right logging settings), I followed the de-sensitization that 500 [handler employs](https://github.com/dimagi/commcare-hq/blob/master/corehq/util/log.py#L57)